hw/top_earlgrey/data/ip/chip_uart_testplan.hjson

```diff
@@ -0,0 +1,252 @@
+                 - Check if there is tx_empty interrupt after transmission of one packet
+                 - Clear the rx_empty interrupt
+                 - Repeat the sequence with 10 packets in a single transaction
+            '''
+      stage: V1
+      si_stage: SV1
+      lc_states: ["TEST_UNLOCKED", "PROD"]
+      features: ["UART.RX_EN", "UART.TX_EN", "UART.RX_WATERMARK_INTERRUPT", "UART.TX_WATERMARK_INTERRUPT",
+                 "UART.BAUD_RATE_CONTROL"]
+      tests: ["chip_sw_uart_tx_rx_smoke"]
+    },
+    {
+      name: chip_sw_uart_tx_rx_parity_smoke
+      desc: '''Verify the transfer of data from UART with parity enabled
+
+               - Powerup OT and enable UART IO.
+               - Calculate CONTROL.NCO using programmers model and update the register
+               - Enable parity in UART
+               Receive sequence:
+                 - Set FIFO_CTRL.RXILVL to 1 so that one character can be received
+                 - Enable RX
+                 - Transfer a single packet from testbench to UART IO with parity bit included
+                 - Check if there is any interrupt and if the source is from UART
+                 - Check if FIFO_STATUS.RXLVL is 1
+                 - Clear the rx_watermark interrupt
+                 - Repeat the sequence with 10 packets in a single transaction
+
+               Transmit sequence:
+                 - Set FIFO_CTRL.TXILVL to 1 so that one character can be transmitted
+                 - Write a random data to WDATA register
+                 - Enable TX
+                 - Monitor UART IO for transmission of data packet
+                 - Check if the packet framing and parity is correct
+                 - Check if there is any interrupt and if the source is from UART
+                 - Check if there is tx_empty interrupt after transmission of one packet
+                 - Clear the rx_empty interrupt
+                 - Repeat the sequence with 10 packets in a single transaction
+            '''
+      stage: V1
+      si_stage: SV1
+      lc_states: ["TEST_UNLOCKED", "PROD"]
+      features: ["UART.RX_EN", "UART.TX_EN", "UART.RX_WATERMARK_INTERRUPT",
+                 "UART.PARITY_ODD", "UART.PARITY_EN", "UART.BAUD_RATE_CONTROL"]
+      tests: ["chip_sw_uart_tx_rx_parity_smoke", "chip_sw_uart_tx_rx"]
+    },
+    // Feature tests
+    {
+      name: chip_sw_uart_loopback
+      desc: ''' Verify the loopback feature of UART
+
+                - Powerup OT and enable UART IO
+                - Calculate CONTROL.NCO using programmers model and update the register
+                - Enable transmission and reception by setting CONTROL.RX and CONTROL.TX
+                - Transmit random bit pattern from testbench to UART
+                - After transmitting each bit wait for one clock period and check the received bit on UART IO
+                - Check if the bit is same as transmitted in earler cycle
+            '''
+      stages: V2
+      si_stage: SV2
```

Comment:
I don't see a reason that this needs to be a "smoke" test. Maybe SV3? Similarly in tests below.

Reply:
Update to SV3